modules/backends/TestBackend.js

```diff
@@ -3,12 +3,35 @@
 export default class TestBackend {
   constructor(manager) {
     this.manager = manager;
+
     const flux = manager.getFlux();
     this.actions = flux.getDragDropActions();
+
+    this.context = manager.getContext();
+  }
+
+  simulateDrop(src, dest) {
+    const source = this.manager.getSource(src);
+    const target = this.manager.getTarget(dest);
+
+    if (!this.manager.getContext().isDragging()) {
+      throw `Can't drop what is not being dragged`;
+    }
+
+    if (target) {
+      const data = target.drop();
+
+      if (typeof data !== 'null') { // let the parent have it?
```

Comment:
dnd-core shouldn't *know* anything about bubbling. Some backends may want to give several drop targets “to choose from”, other backends may not even be able to do so. Therefore I think it's sensible to change `simulateDrop` to accept *an array* of drop targets.

Now that I think of it, if we implement enter/leave tracking, we will *already know* which drop zones we have entered. Therefore, we won't need to rely on event bubbling: we can just ask “entered” drop zones one by one.

I need to think some more about this.